src/app/loadout/mod-utils.ts

```diff
@@ -325,62 +327,85 @@ function createOrderedAssignmentResults(
   bucketSpecificAssignments: PluggableInventoryItemDefinition[],
   bucketIndependentAssignments: PluggableInventoryItemDefinition[]
 ) {
-  const results: {
+  const pluggingActions: {
     socketIndex: number;
     mod?: PluggableInventoryItemDefinition;
     // This will be negative if we are recovering used energy back by swapping in a cheaper mod
     energyChange: number;
   }[] = [];
 
-  const mods = [...bucketIndependentAssignments, ...bucketSpecificAssignments];
+  const modsToInsert = [...bucketIndependentAssignments, ...bucketSpecificAssignments];
 
-  const modIndexes =
+  const armorModIndexes =
     item.sockets?.categories.find(
       (category) => category.category.hash === SocketCategoryHashes.ArmorMods
     )?.socketIndexes || [];
-  const modSockets = getSocketsByIndexes(item.sockets!, modIndexes);
+  const existingModSockets = getSocketsByIndexes(item.sockets!, armorModIndexes);
 
-  for (const mod of mods) {
-    // Check to see if its already socketed somewhere
-    let sortedSocketIndex = modSockets.findIndex(
+  for (const mod of modsToInsert) {
+    // If it's already plugged somewhere, that's the slot we want to "plug it into"
+    let destinationSocketIndex = existingModSockets.findIndex(
       (socket) => socket.plugged?.plugDef.hash === mod.hash
     );
-    if (sortedSocketIndex < 0) {
-      // Otherwise find the first socket it can fit into
-      sortedSocketIndex = modSockets.findIndex(
+    // If it wasn't found already plugged, find the first socket it can fit into
+    if (destinationSocketIndex === -1) {
+      destinationSocketIndex = existingModSockets.findIndex(
         (socket) => socket.plugged?.plugDef.plug.plugCategoryHash === mod.plug.plugCategoryHash
       );
     }
 
-    // This should always be the case, if its not then something is seriously wrong
-    if (sortedSocketIndex > 0) {
-      const pluggedCost =
-        modSockets[sortedSocketIndex].plugged?.plugDef.plug.energyCost?.energyCost || 0;
-      const modCost = mod.plug.energyCost?.energyCost || 0;
-      const energyChange = modCost - pluggedCost;
-      results.push({ socketIndex: modSockets[sortedSocketIndex].socketIndex, mod, energyChange });
-      modSockets.splice(sortedSocketIndex, 1);
+    // If a destination socket couldn't be found for this plug, something is seriously wrong
```

Comment:
Maybe not seriously wrong - imagine your mod loadout contains raid mods but the items you're applying to don't have that slot.

Reply:
That should be handled previously in the mod assignment algorithm. At this stage we should just have a set of mods we know can be assigned to the item.

I split the assignment algorithm and this into seperate functions.